langchain-core/src/messages/content_blocks.ts

```diff
@@ -0,0 +1,77 @@
+export interface BaseDataContentBlock {
+  mime_type?: string;
+}
+
+export interface URLContentBlock extends BaseDataContentBlock {
+  type: "image" | "audio" | "file";
+  source_type: "url";
+  url: string;
+}
+
+export interface Base64ContentBlock extends BaseDataContentBlock {
+  type: "image" | "audio" | "file";
+  source_type: "base64";
+  data: string;
+}
+
+export interface PlainTextContentBlock extends BaseDataContentBlock {
+  type: "file";
+  source_type: "text";
+  text: string;
+}
+
+export interface IDContentBlock extends BaseDataContentBlock {
+  type: "image" | "audio" | "file";
+  source_type: "id";
+  id: string;
+}
+
+export type DataContentBlock =
+  | URLContentBlock
+  | Base64ContentBlock
+  | PlainTextContentBlock
+  | IDContentBlock;
+
+export function isDataContentBlock(
+  content_block: object
+): content_block is DataContentBlock {
+  return (
+    typeof content_block === "object" &&
+    content_block !== null &&
+    "source_type" in content_block &&
+    (content_block.source_type === "url" ||
+      content_block.source_type === "base64" ||
+      content_block.source_type === "text" ||
+      content_block.source_type === "id")
+  );
+}
+
+export function convertToOpenAIImageBlock(
+  content_block: object
```

Comment:
since it's exported we should capture the type requirements here

```suggestion
  content_block: DataContentBlock
```